src/util/key.rs

```diff
@@ -109,15 +109,19 @@ impl PublicKey {
         }
     }
 
-    /// Returns bitcoin 160-bit hash of the public key
-    pub fn pubkey_hash(&self) -> PubkeyHash {
+    fn with_serialized<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
         if self.compressed {
-            PubkeyHash::hash(&self.inner.serialize())
+            f(&self.inner.serialize())
         } else {
-            PubkeyHash::hash(&self.inner.serialize_uncompressed())
+            f(&self.inner.serialize_uncompressed())
         }
     }
 
+    /// Returns bitcoin 160-bit hash of the public key
+    pub fn pubkey_hash(&self) -> PubkeyHash {
+        self.with_serialized(PubkeyHash::hash)
```

Comment:
I like this PR but this line doesn't read that clearly to me (just flagging it to add context to my function re-name suggestion).

Reply:
I think it's pretty clear --- what would you rename?

Reply:
Ah you suggest `serialize_and_call` .. yeah, I could get behind that.

Reply:
I've actually seen several crates use `with_` convention when a function accepts closure. E.g.: https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with_key Thus I believe this is considered idiomatic even though you didn't see this one yet.

Reply:
Oh no worries, glad to learn it then. Thanks man.